x/group/keeper/msg_server.go

```diff
@@ -779,18 +749,24 @@ func (k Keeper) Exec(goCtx context.Context, req *group.MsgExec) (*group.MsgExecR
 		_, err = k.doExecuteMsgs(ctx, k.router, proposal, addr)
 		if err != nil {
 			proposal.ExecutorResult = group.PROPOSAL_EXECUTOR_RESULT_FAILURE
-			proposalType := reflect.TypeOf(proposal).String()
-			logger.Info("proposal execution failed", "cause", err, "type", proposalType, "proposalID", id)
+			k.Logger(ctx).Info("proposal execution failed", "cause", err, "proposalID", id)
 		} else {
 			proposal.ExecutorResult = group.PROPOSAL_EXECUTOR_RESULT_SUCCESS
 			flush()
 		}
 	}
 
 	// Update proposal in proposalTable
-	res, err := storeUpdates()
-	if err != nil {
-		return nil, err
+	// If proposal has successfully run, delete it from state.
+	if proposal.ExecutorResult == group.PROPOSAL_EXECUTOR_RESULT_SUCCESS {
+		if err := k.pruneProposal(ctx, proposal.Id); err != nil {
+			return nil, err
+		}
+	} else {
+		store := ctx.KVStore(k.key)
+		if err := k.proposalTable.Update(store, id, &proposal); err != nil {
+			return nil, err
+		}
 	}
 
 	err = ctx.EventManager().EmitTypedEvent(&group.EventExec{
```

Comment:
Currently, if the proposal execution has an error, we simply do logger.Info to log the error.

How about adding a field `Log` to `EventExec`? empty is the result is SUCCESS, and contains the error log if the result is FAILURE

Reply:
SGTM

Reply:
done